src/org/traccar/database/DeviceManager.java

```diff
@@ -104,15 +104,35 @@ public Device getDeviceByPhone(String phone) {
         return getItems(getAllItems());
     }
 
+    public Set<Long> getAllUserItems(long userId) {
+        return Context.getPermissionsManager().getDevicePermissions(userId);
+    }
+
     @Override
     public Set<Long> getUserItems(long userId) {
         if (Context.getPermissionsManager() != null) {
-            return Context.getPermissionsManager().getDevicePermissions(userId);
+            Set<Long> result = new HashSet<>();
+            for (long deviceId : Context.getPermissionsManager().getDevicePermissions(userId)) {
+                Device device = Context.getIdentityManager().getById(deviceId);
```

Comment:
Do we really need `getIdentityManager` here? You are already in `DeviceManager`.